src/delegate.ts

```diff
@@ -0,0 +1,38 @@
+import { Handle } from 'dojo-core/interfaces';
+import on, { ExtensionEvent } from 'dojo-core/on';
+
+export default function delegate(target: HTMLElement, selector: string, type: string, listener: (event: UIEvent) => void): Handle;
+export default function delegate(target: HTMLElement, selector: string, type: ExtensionEvent, listener: (event: UIEvent) => void): Handle;
+export default function delegate(target: HTMLElement, selector: string, type: (string | ExtensionEvent)[], listener: (event: UIEvent) => void): Handle;
+export default function delegate(target: HTMLElement, selector: string, type: any, listener: (event: UIEvent) => void): Handle {
+	function matches(target: EventTarget) {
+		let node = <any> target;
+		let matchMethod = (function() {
```

Comment:
We shouldn't have to execute this logic every single time each delegated event handler runs.  This looks like a prime candidate for a `has` feature test to me (which should go in `src/has` which doesn't exist yet).  It should be fine for the feature test to return the method name.